### ISSUE TITLE
Disabling "Lock Feature Control Register"  in HII has no effect after reset. (Bugzilla Bug 1305)

### ISSUE THREAD

**Issue by tianocore-issues** [tianocore/edk2#7616]
This issue was created automatically with [bugzilla2github](https://github.com/makubacki/bugzilla2github/tree/bz2gh_tianocore)
# Bugzilla Bug 1305
Date: 2018-11-07T04:02:45+00:00
From: @niruiyu
To:   @niruiyu
CC:   @lersek

See also:     https://bugzilla.tianocore.org/show_bug.cgi?id=2040
Last updated: 2019-09-11T13:42:24+00:00

**Comment by tianocore-issues** [comment]
## Comment 5520
Date: 2018-11-07 04:02:45 +0000
From: @niruiyu

- Industry Specification: ---
- Target OS: ---
- Bugzilla Assignee(s): @niruiyu

Disabling &quot;Lock Feature Control Register&quot; in HII.
Reset
Check the MSR_IA32_FEATURE_CONTROL (0x3A) bit [0]
The bit is still 1.

**Comment by tianocore-issues** [comment]
## Comment 5521
Date: 2018-11-07 04:05:47 +0000
From: @niruiyu

fixed @ 1ed6498c4

**Comment by tianocore-issues** [comment]
## Comment 5578
Date: 2018-11-13 22:03:26 +0000
From: @niruiyu

reopen this issue because for the sake of security, FEATURE_CONTROL.Lock bit is better to be set always.

Revert 1ed6498c4 in below commit:
SHA-1: 85588389222a3636baf0f9ed8227f2434af4c3f9
* UefiCpuPkg/CommonFeature: Always set FEATURE_CONTROL.Lock

**Comment by tianocore-issues** [comment]
## Comment 9319
Date: 2019-08-27 19:42:55 +0000
From: @lersek

Ray: has this been ultimately fixed in bug 2040?

Should we close this BZ as a duplicate of bug 2040?

Thanks.

**Comment by tianocore-issues** [comment]
## Comment 9579
Date: 2019-09-10 20:26:33 +0000
From: @niruiyu

(In reply to Laszlo Ersek from comment #3)
&gt; Ray: has this been ultimately fixed in bug 2040?
&gt; 
&gt; Should we close this BZ as a duplicate of bug 2040?
&gt; 
&gt; Thanks.

This is not related to bug 2040.
From the security perspective, it&apos;s not good to keep the Lock bit clear even controlled through HII setup.
So I will just close this.

**Comment by tianocore-issues** [comment]
## Comment 9616
Date: 2019-09-11 11:48:40 +0000
From: @lersek

Ray,

I don&apos;t understand why we should close this bug as &quot;FIXED&quot;. Where and when has it been fixed?

Comment 1 names a patch that supposedly fixed the problem. But then comment 2 identifies a patch that reverted the patch from comment 1. Based on that, we are back to square one; the problem is not fixed, it exists just like when this BZ was reported. That&apos;s why I asked in comment 3 if we should close this as a duplicate.

If we have *genuinely* fixed this issue, then please identify the commit hash (SHA1) of the resolution. If we have not fixed the issue, but also do not *plan* to fix it, then we should change the &quot;FIXED&quot; resolution to &quot;WONTFIX&quot;.

Right now the outcome of this BZ is very confusing.

Thanks
Laszlo

**Comment by tianocore-issues** [comment]
## Comment 9619
Date: 2019-09-11 13:36:37 +0000
From: @niruiyu

Laszlo,
The bug was found by QA and I thought it was a bug that needed fix.
But talking with the security guy, he suggested the always locking was intended so the security-sensitive register cannot be unblocked by user mistake.
The correct fix is in Intel&apos;s close source code to not expose that capability (keep the lock bit open).

**Comment by tianocore-issues** [comment]
## Comment 9621
Date: 2019-09-11 13:42:24 +0000
From: @lersek

Thank you, understood. In this case, the proper resolution for the current BZ, is &quot;INVALID&quot;. That&apos;s because the open source edk2 tree does not need any fix -- it is not affected by the problem description. I&apos;m updating the resolution accordingly.
